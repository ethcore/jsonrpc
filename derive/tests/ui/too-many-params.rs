extern crate serde;
extern crate jsonrpc_core;
#[macro_use]
extern crate jsonrpc_derive;

use jsonrpc_core::Result;

#[rpc]
pub trait Rpc {
	/// Has too many params
	#[rpc(name = "tooManyParams")]
	fn to_many_params(
        &self,
        _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64,
        _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64,
    ) -> Result<String>;
}

fn main() {}