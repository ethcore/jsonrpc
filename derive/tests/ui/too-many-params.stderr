error: Maximum supported number of params is 16
  --> $DIR/too-many-params.rs:10:2
   |
10 |       /// Has too many params
   |  _____^
11 | |     #[rpc(name = "tooManyParams")]
12 | |     fn to_many_params(
13 | |         &self,
14 | |         _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64,
15 | |         _: u64, _: u64, _: u64, _: u64, _: u64, _: u64, _: u64,
16 | |     ) -> Result<String>;
   | |________________________^

error: aborting due to previous error

